### PR TITLE
166 - Implement useOneOffExperiment hook for one-off experiment dataset downloads

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number

Closing #166

## Purpose/Implementation Notes
Implemented the hook `useOneOffExperiment` which manages the file download for one-off experiment dataset. 

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

**NOTE:** The`Notification` and `[dataset_id]` files are included in the files changed count but they can be safely ignored. They will be removed after the previous base branch is merged into `dev` (this previous base branch hasn't  yet pulled the most recent `dev` changes).

Commits in this PR:
- af9193270cb08ee1eb8bf0914b6f41de626eae04
- 43b0c47ce639d7fb3f415fefacb4ed181818194c
- 6d8a3bbd01cdccae2fc5b00b068b44ac6a72f5aa
- aeb7e7b4dc97f999179190d132cc9edbd8660409

### CODE REVIEW

File: src/hooks/useOneOffExperiment.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    setProcessingExperiments([
+      ...processingExperiments,
+      { accessionCode, datasetId }
+    ])
+  }
+
+  const getProcessingExperiment = (accessionCode) => {
+    if (!accessionCode || !processingExperiments.length) return null
+
+    const [experiment] = processingExperiments.filter(
```
Comment (davidsmejia) [inline src/hooks/useOneOffExperiment.js:46]: Can we use Array.prototype.find here?

**Review by davidsmejia — CHANGES_REQUESTED** [review]
One change requested otherwise looking good

**Comment by nozomione** [comment]
@davidsmejia , please review PR https://github.com/AlexsLemonade/refinebio-web/pull/218 first before reviewing this PR. 

🗒️  I've added a note about this to the **Issue Number** section and also left a comment (https://github.com/AlexsLemonade/refinebio-web/pull/205#issuecomment-1770991338) in previous PR https://github.com/AlexsLemonade/refinebio-web/pull/205 (which was already reviewed & approved), thank you.

**Review by davidsmejia — APPROVED** [review]
I think tracking both `startTimer` and `timerId` is redundant but we can address that at another time. Otherwise this looks good.

**Pull request merged**